src/v2.rs

```
@@ -166,7 +194,11 @@ impl Party {
             return Err(DkgError::BadIds(bad_ids));
         }
 
+        /* MM:
+        Move simple test to the top, fail fast
+         */
```

Suggestion: move this simple test to the top of the function body, so it can fail fast.